### PR TITLE
[LTD-751] Add middleware to validate `return_to` links

### PR DESCRIPTION
**Pull request by alixedi**
The `return_to` parameter is used to set back-links in the application journey.

This adds a middleware to the application that validates if the `return_to` parameter is:

1. A valid URL
2. A relative URL

Would recommend reviewing commit-wise.

### CODE REVIEW

File: unit_tests/core/test_middleware.py [hunk 1]
```diff
@@ -1,13 +1,37 @@
...
+        ("?return_to=hello", status.HTTP_200_OK),
+        ("?return_to=hello/", status.HTTP_200_OK),
+        ("?return_to=/hello", status.HTTP_200_OK),
+        ("?return_to=/hello/", status.HTTP_200_OK),
+        ("?return_to=http://example.com", status.HTTP_400_BAD_REQUEST),
+        ("?return_to=http://example.com/", status.HTTP_400_BAD_REQUEST),
+        ("?return_to=https://example.com/", status.HTTP_400_BAD_REQUEST),
+        ('?return_to=javascript:alert("hello!")', status.HTTP_400_BAD_REQUEST),
+        # urlparse strips off the leading // and we are left with //example.com
+        ("?return_to=////example.com", status.HTTP_200_OK),
```
Comment (r4vi) [inline unit_tests/core/test_middleware.py:30]: is this OK? wouldn't it mean the back button would take you to //example.com?

Comment (alixedi) [reply unit_tests/core/test_middleware.py:30]: I am having to dig deeper for protocol-relative URLs b/c `urlparse` doesn't seem to detect them properly.

Comment (r4vi) [reply unit_tests/core/test_middleware.py:30]: yes sorry I used an intentionally sneaky one there with the 4 `/`

Comment (r4vi) [reply unit_tests/core/test_middleware.py:30]: yes sorry I used an intentionally sneaky one there with the 4 `/`

Comment (alixedi) [reply unit_tests/core/test_middleware.py]: Not at all. This was very useful 👏

**Review by r4vi — APPROVED** [review]

**Pull request merged**